src/NzbDrone.Core/Notifications/SendGrid/SendGridSettings.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Notifications.SendGrid
+{
+    public class SendGridSettingsValidator : AbstractValidator<SendGridSettings>
+    {
+        public SendGridSettingsValidator()
+        {
+            RuleFor(c => c.ApiKey).NotEmpty();
+            RuleFor(c => c.From).NotEmpty().EmailAddress();
+            RuleFor(c => c.Recipients).NotEmpty();
+            RuleForEach(c => c.Recipients).NotEmpty().EmailAddress();
+        }
+    }
+
+    public class SendGridSettings : IProviderConfig
+    {
+        private static readonly SendGridSettingsValidator Validator = new SendGridSettingsValidator();
+
+        public SendGridSettings()
+        {
+            BaseUrl = "https://api.sendgrid.com/v3/";
+            Recipients = new string[] { };
+        }
+
+        public string BaseUrl { get; set; }
+
+        [FieldDefinition(1, Label = "API Key", HelpText = "The API Key generated by SendGrid")]
```

Comment:
Can we get a `HelpLink` to the appropriate send grid page to get this?

Reply:
Ah, yes. Pointing it to creating api keys on their docs.